packages/hydrooj/src/handler/judge.ts

```diff
@@ -248,6 +248,7 @@ export class JudgeConnectionHandler extends ConnectionHandler {
         if (!rdoc) return;
 
         const rid = rdoc._id.toHexString();
+        if (this.tasks[rid]) return;
```

Comment:
Currently no idea how to handle this.

Having multiple task for the same record are usually not allowed so in model/record#L83 the existing task will be cancelled before adding a new task, however it's possible and should be allowed for hack rejudges.

Reply:
What I imaged was someone spamming the `rejudge` button and causing multiple tasks got inserted into task. Indeed, there's logic to ensure there's only one task in the `task` collection but it could still conflict with running ones since previous one got deleted before executing.

Reply:
ah that's true. but in this case I think we should drop the older task, not the newer one?

for example someone submitted a code that might take a few minutes to judge, and when it's running the admin found something incorrect with testdata and clicked the rejudge all button. You definitely want that submission rejudged instead of dropped.

Reply:
also I suggest to do this in another commit.

Reply:
If we want to implement logic to cancel running task, we need collaboration between the web and the judge. For example, sending out a cancel request from web to judge to inform it to cancel the execution. Otherwise, the subsequent `next` / `end` message will mess up the current execution. Or, we may choose to disable the `rejudge` / `hack` functionality when it is being executing (in status `fetched`, `compiling`, `judging`).

Reply:
to implement this we have to add taskId along with (or insteadof as a quick patch) recordId and use that as key in communication. also consider maybe two tasks are assigned to different consumer :(

Reply:
In this case, it would be like an exclusive lock by task for records. For example, add `taskId` field to `record` when fetched and only permits edit to record when `taskId` matches. Then, how to compliant with old version of judge where `taskId` not presented...

Reply:
Ah, that's too much to consider for this edge case. I'll remove this change and focus on the docker file change only.